ci/build_docs.sh

```diff
@@ -19,29 +19,29 @@ rapids-print-env
 rapids-logger "Downloading artifacts from previous jobs"
 CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
 PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
-VERSION_NUMBER="23.08"
 
 rapids-mamba-retry install \
   --channel "${CPP_CHANNEL}" \
   --channel "${PYTHON_CHANNEL}" \
   rmm librmm
 
-# Build CPP docs
-rapids-logger "Build Doxygen docs"
+export RAPIDS_VERSION_NUMBER="23.08"
+export RAPIDS_DOCS_DIR=${RAPIDS_DOCS_DIR:-"${PWD}/documentation"}
```

Comment:
```suggestion
export RAPIDS_DOCS_DIR="$(mktemp -d)"
```
In hindsight, we can just make this variable a temporary directory.

I originally thought it would be useful to set this environment variable at the shared-workflow level, but the variable's value is only ever used in this particular workflow step.

Therefore we don't need to set it at the workflow level and can simply set it in this script.